src/ragger/navigator/interface.py

```diff
@@ -0,0 +1,288 @@
+
+    def navigate_until_snap(self,
+                            navigate_instruction: NavigationInstruction,
+                            validation_instruction: NavigationInstruction,
+                            path: Path,
+                            test_case_name: Path,
+                            start_img_idx: int = 0,
+                            last_img_idx: int = 0,
+                            take_snaps: bool = True,
+                            timeout: int = 30,
+                            crop_first: Crop = None,
+                            crop_last: Crop = None) -> int:
+        """
+        Navigate until snapshot is found.
+
+        This method may be left void on backends connecting to physical devices,
+        where a physical interaction must be performed instead.
+        This will prevent the instrumentation to fail (the void method won't
+        raise `NotImplementedError`), but the instrumentation flow will probably
+        get stuck (on further call to `receive` for instance) until the expected
+        action is performed on the device.
+
+        :param navigate_instruction: Navigation instruction to be performed until last snapshot is found.
+        :type navigate_instruction: NavigationInstruction
+        :param validation_instruction: Navigation instruction to be performed once last snapshot is found.
+        :type validation_instruction: NavigationInstruction
+        :param path: Absolute path to the snapshots directory.
+        :type path: Path
+        :param test_case_name: Relative path to the test case snapshots directory (from path).
+        :type test_case_name: Path
+        :param start_img_idx: Index of the first snapshot of the navigation flow.
+        :type start_img_idx: int
+        :param last_img_idx: Index of the snapshot to look for when navigating.
+        :type last_img_idx: int
+        :param take_snaps: Take temporary snapshots of the screen displayed when navigating.
+        :type take_snaps: bool
+        :param timeout: Timeout of the navigation loop if last snapshot is not found.
+        :type timeout: int
+        :param crop_first: Crop (left, upper, right or lower pixels) first snapshot image for comparison
+                           (useful if using a generic snapshot).
+        :type crop_first: Crop
+        :param crop_last: Crop (left, upper, right or lower pixels) last snapshot image for comparison
+                          (useful if using a generic snapshot).
+        :type crop_last: Crop
+
+        :return: img_idx
+        :rtype: int
+        """
+
+        if not isinstance(self._backend, SpeculosBackend):
+            # When not using Speculos backend, taking snapshots is not possible
+            # therefore comparison is not possible too.
+            # TODO request user to interact with the device.
+            return 0
+
+        snaps_golden_path = self._get_snaps_dir_path(path, test_case_name, True)
+        if not snaps_golden_path.is_dir():
```

Comment:
Same remark as before. It's reasonnable to ask `self._get_snaps_dir_path` to assert if the directory exists or not (and is a directory). It could be optionnable through a keyword argument if we don't want to check everytime.